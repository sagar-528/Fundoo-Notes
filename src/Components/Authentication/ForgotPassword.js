import React from 'react'
import { View, Text, TouchableWithoutFeedback, Keyboard, TouchableOpacity, TextInput } from 'react-native';
import * as Animatable from 'react-native-animatable';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import GlobalCss from '../../Styles/GlobalCss';
import { Dialog, Portal, Button, Paragraph, Provider } from 'react-native-paper';
import UserServices from '../../../Service/UserServices'
 

class ForgetPassword extends React.Component {

    constructor(props) {
        super(props)
    
        this.state = {
            email : '',
            invalidEmail : false,
            emailEmpty : false,
            visible : false
        }
    }

textInputChangeEmail = (val) =>{
    this.setState({
        email: val,
        invalidEmail : false,
        emailEmpty : false,
    })
}

 resetPasswordHandler = () => {
    const {onPress} = this.props
    if(this.state.email != '') {
        UserServices.forgotPassword(this.state.email)
                .then( () => {
                     this.setState({
                        visible : true
                    })
                })
                .catch( error => {
                    if(error == 'Email not Found') {
                         this.setState({
                            invalidEmail : true
                        })
                    }
                })
    }
    else {
        if(this.state.email == '') {
             this.setState({
                emailEmpty : true
            })
        }
    }
    // onPress();
}

hideDialog =  () => {
    const {onDismiss} = this.props
     this.setState({
        visible : false
    })
    // onDismiss();
}

handleDialogButton = () => {
    const {onPress} = this.props
    this.props.navigation.navigate('SignIn')
    // onPress();
}


render(){
  return(
      <Provider>
    <TouchableWithoutFeedback onPress = {() => {
      Keyboard.dismiss();
    }}>
    <View style = {GlobalCss.container}>
    
    <View style = {GlobalCss.header}>
        <FontAwesome5 
            name = "foursquare"
            color = "#fff"
            size = {50}
        />
      <Text style = {GlobalCss.text_header}>FundooNotes</Text>
    </View>

    <Animatable.View 
    animation = "fadeInUpBig"
    style = {GlobalCss.footer}
    >
    <Text style = {[GlobalCss.text_footer,{marginTop: 30}]}>Email-Id</Text>
    
        <View style = {GlobalCss.action}>
            <FontAwesome5 
                name = "envelope"
                size = {20}
                color = "black"
            />
            <TextInput 
                placeholder = "Your Email-Id"
                placeholderTextColor = "black"
                style = {GlobalCss.textInput}
                onChangeText = {this.textInputChangeEmail}
            />
        </View>
    
        <View>    
            <Text style = {GlobalCss.text_error_style}>
                {(this.state.emailEmpty) ? 'required..' : (this.state.invalidEmail) ? 'Email not Found..' : null}
            </Text>
        </View>     

        <View style = {{alignItems: 'center', marginTop: 20}}>
        <TouchableOpacity
            style = {[GlobalCss.signIn, {
                borderColor: '#009387',
                borderWidth: 1,
                marginTop: 15
            }]}
            onPress = { this.resetPasswordHandler }
        >
            <Text style = {[GlobalCss.textSign, {
                color: 'black'
            }]}>Reset Password</Text>
        </TouchableOpacity>
        </View>
        <Portal>
            <Dialog visible={this.state.visible} onDismiss={this.hideDialog}>
                <Dialog.Content>
                    <Paragraph style = {{fontSize : 20, paddingTop : 20}}>Mail is send to your Email-Id.</Paragraph>
                </Dialog.Content>
                    <Dialog.Actions>
                        <Button onPress={this.handleDialogButton}>Ok</Button>
                    </Dialog.Actions>
            </Dialog>
        </Portal>
    </Animatable.View>
    </View>
    </TouchableWithoutFeedback>
    </Provider>
  )};
};

export default ForgetPassword
