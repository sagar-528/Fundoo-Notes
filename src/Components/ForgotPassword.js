import React from 'react'
import { View, Text, TouchableWithoutFeedback, Keyboard, TouchableOpacity, TextInput } from 'react-native';
import * as Animatable from 'react-native-animatable';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import GlobalCss from '../Styles/GlobalCss';
import UserServices from '../../Service/UserServices'

class ForgetPassword extends React.Component {

    constructor(props) {
        super(props)
    
        this.state = {
            email : '',
            invalidEmail : false,
            emailEmpty : false
        }
    }

textInputChangeEmail = (val) =>{
    this.setState({
        email: val,
        invalidEmail : false,
        emailEmpty : false,
    })
}

 resetPasswordHandler = () => {
    if(this.state.email != '') {
        UserServices.forgotPassword(this.state.email)
            .then(user => this.props.navigation.push('SignIn'))
            .catch(error => {
                if(error == 'Email not Found') {
                    this.setState({
                        invalidEmail : true
                    })
                }
            })
    }
    else {
        if(this.state.email == '') {
             this.setState({
                emailEmpty : true
            })
        }
    }
}

render(){
  return(
    <TouchableWithoutFeedback onPress = {() => {
      Keyboard.dismiss();
    }}>
    <View style = {GlobalCss.container}>
    
    <View style = {GlobalCss.header}>
        <FontAwesome5 
            name = "user-circle"
            color = "#fff"
            size = {50}
        />
      <Text style = {GlobalCss.text_header}>FundooNotes</Text>
    </View>

    <Animatable.View 
    animation = "fadeInUpBig"
    style = {GlobalCss.footer}
    >
    <Text style = {[GlobalCss.text_footer,{marginTop: 30}]}>Email-Id</Text>
    
    <View style = {GlobalCss.action}>
        <FontAwesome5 
            name = "envelope"
            size = {20}
            color = "black"
        />
        <TextInput 
            placeholder = "Your Email-Id"
            placeholderTextColor = "black"
            style = {GlobalCss.textInput}
            onChangeText = {this.textInputChangeEmail}
        />
    </View>
    
    <View>    
        <Text style = {GlobalCss.text_error_style}>
            {(this.state.emailEmpty) ? 'required..' : (this.state.invalidEmail) ? 'Email not Found..' : null}
        </Text>
    </View>     

        <View style = {{alignItems: 'center', marginTop: 20}}>
        <TouchableOpacity
            style = {[GlobalCss.signIn, {
                borderColor: '#009387',
                borderWidth: 1,
                marginTop: 15
            }]}
            onPress = { this.resetPasswordHandler }
        >
            <Text style = {[GlobalCss.textSign, {
                color: 'black'
            }]}>Reset Password</Text>
        </TouchableOpacity>
    </View>
    </Animatable.View>
    </View>
    </TouchableWithoutFeedback>
  )};
};

export default ForgetPassword
