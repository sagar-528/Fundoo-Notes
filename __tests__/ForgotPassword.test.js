import React from 'react';
import {shallow} from 'enzyme';
import ForgotPassword from '../src/Components/ForgotPassword'

describe('test Forgot Password', () => {
    
    it('should match to snapshot', () => {
        const component = shallow(<ForgotPassword/>)
        expect(component).toMatchSnapshot();
    })

    it('test when email provided in textinput should update email state', async () => {
        const component = shallow(<ForgotPassword />)
        expect(component.instance().state.email).toBe('')
        component.instance().textInputChangeEmail('gupta.sagar528@gmail.com')
        expect(component.instance().state.email).toBe('gupta.sagar528@gmail.com')
    })

    it('test onPress event of reset password button when all textinput empty it will update the stateEmpty state to true ', async () => {
        const onPressEvent = jest.fn();
        const component = shallow(<ForgotPassword onPress = {onPressEvent}/>)
        const instance = component.instance();
        await instance.resetPasswordHandler();
        expect(onPressEvent).toHaveBeenCalled();
        expect(instance.state.emailEmpty).toBe(true)
    })
    
})