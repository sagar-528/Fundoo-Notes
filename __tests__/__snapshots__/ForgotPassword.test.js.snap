// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Forgot Password should match to snapshot 1`] = `ShallowWrapper {}`;
